File: src/main/java/org/example/twopointers/TwoSumII.java
package org.example.twopointers;

public class TwoSumII {
    public int[] twoSum(int[] numbers, int target) {
        int idx1=0, idx2=numbers.length-1;

        for (int i=0; i<numbers.length; i++) {
            if (numbers[idx1] + numbers[idx2] == target) {
                return new int[]{++idx1, ++idx2};
            } else if (numbers[idx1] + numbers[idx2] < target) {
                idx1++;
            } else if (numbers[idx1] + numbers[idx2] > target) {
                idx2--;
            }
        }

        return new int[]{++idx1, ++idx2};
    }
}

File: src/main/java/org/example/twopointers/TwoSumII.py
class Solution(object):
    def twoSum(self, numbers, target):
        idx1, idx2 = 0, len(numbers) - 1

        for i in range(len(numbers)):
            if numbers[idx1] + numbers[idx2] == target:
                return [idx1 + 1, idx2 + 1]
            elif numbers[idx1] + numbers[idx2] < target:
                idx1 += 1
            elif numbers[idx1] + numbers[idx2] > target:
                idx2 -= 1
        return [idx1 + 1, idx2 + 1]
